### PR TITLE
refactoring EmailClient and related unit tests with MailKit and RazorLight

### PR DESCRIPTION
**Pull request by Steve3029**

### CODE REVIEW

File: src/DevYeah.LMS/DevYeah.LMS.Business/AccountService.cs [hunk 1]
```diff
@@ -298,31 +296,25 @@ private string BuildPasswordRecoveryMail(Account account)
             return content;
         }
 
-        private bool SendEmail(string emailAddress, string subject, string content)
+        private void SendEmail(Action logic, int maxRetryCounter, Action logImportant = null, Action logError = null)
```
Comment (unruledboy) [inline src/DevYeah.LMS/DevYeah.LMS.Business/AccountService.cs]: Please move it into the current EmailClient class

**Comment by Steve3029** [comment]
I have refactored EmailClient class and a part of AccountService class. If there is any null exception issue that I do not fix, please do me a favor to mark it.

File: src/DevYeah.LMS/DevYeah.LMS.Business/EmailClient.cs [hunk 2]
```diff
@@ -19,31 +20,58 @@ public EmailClient(IOptions<EmailSettings> emailSettings)
...
         {
             if (string.IsNullOrWhiteSpace(email)) throw new ArgumentNullException(nameof(email));
+            if (string.IsNullOrWhiteSpace(subject)) throw new ArgumentNullException(nameof(subject));
             if (string.IsNullOrWhiteSpace(content)) throw new ArgumentNullException(nameof(content));
 
             var message = WrappingEmail(email, subject, content);
+            RetryAction(() => Sending(message), _emailSettings.MaxRetryCount);
+        }
+
+        private void Sending(MimeMessage message)
```
Comment (unruledboy) [inline src/DevYeah.LMS/DevYeah.LMS.Business/EmailClient.cs]: Normally a function name should be an action, so `Sending` should be just `Send`. It's fine, it's compile time polymorphism, aka overloads.

File: src/DevYeah.LMS/DevYeah.LMS.Business/EmailClient.cs [hunk 3]
```diff
@@ -19,31 +20,58 @@ public EmailClient(IOptions<EmailSettings> emailSettings)
...
+                {
+                    logImportant?.Invoke();
+                    Thread.Sleep(1000);
+                }
+            } while (loopCounter < maxRetryCounter);
+            if (loopCounter > 1)
+                logError?.Invoke();
+        }
+
         private MimeMessage WrappingEmail(string email, string subject, string content)
```
Comment (unruledboy) [inline src/DevYeah.LMS/DevYeah.LMS.Business/EmailClient.cs]: For `WrappingEmail`, maybe `CreateEmailMessage` better?

File: src/DevYeah.LMS/DevYeah.LMS.Business/AccountService.cs [hunk 4]
```diff
@@ -115,7 +106,6 @@ private ClaimsPrincipal GetClaimsPrincipal(string token)
             }
             catch (Exception)
```
[the remark this answers is not in the document]
Comment (Steve3029) [reply src/DevYeah.LMS/DevYeah.LMS.Business/AccountService.cs]: The method "ValidateToken" might throw some exceptions like ArgumentNullException, SecurityTokenException and etc.

**Pull request merged**